components/prism-pascaligo.js

```diff
@@ -0,0 +1,45 @@
+// Pascaligo is a layer 2 smart contract language for the tezos blockchain
+
+Prism.languages.pascaligo = {
+	'comment': /\(\*[\s\S]+?\*\)|\/\/.*/,
+	'string': [
+		{
+			pattern: /(?:'(?:''|[^'\r\n])*')+|\^[a-z]/i,
```

Comment:
This pattern `(?:'(?:''|[^'\r\n])*')+` as well as the below `"`-version can potentially show exponential backtracking because of the `+`.
Luckily, this is really easy to fix as the patterns `(?:'(?:''|[^'\r\n])*')+` and `'(?:''|[^'\r\n])*'` are equivalent.

```suggestion
			pattern: /'(?:''|[^'\r\n])*'|\^[a-z]/i,
```